readthedocs/config/config.py

```diff
@@ -124,7 +125,9 @@ def _get_display_key(self):
         # Checks for patterns similar to `python.install.0.requirements`
         # if matched change to `python.install[0].requirements` using backreference.
         return re.sub(
-            r'^(python\.install)(\.)(\d+)(\.\w+)$', r'\1[\3]\4', self.key
+            r'^([a-zA-Z_.-]+)\.(\d+)([a-zA-Z_.-]*)$',
+            r'\1[\2]\3',
+            self.key
```

Comment:
Does this need to update the docstring? Seems it supports more than `python.install` now?